core/global/input_manager.gd

```diff
@@ -130,9 +130,9 @@ func _on_gamepad_change(_device: int, _connected: bool) -> void:
 			continue
 
 		logger.debug("Gamepad disconnected: " + gamepad.phys_path)
-		orphaned_gamepads[gamepad.phys] = gamepad
 		# Lock the gamepad mappings so we can alter them.
 		gamepad_mutex.lock()
+		orphaned_gamepads[gamepad.phys] = gamepad
```

Comment:
This is an unnecessary change as it doesn't appear that `orphaned_gamepads` is referenced by anything in a separate thread. Access to it appears to all be on the main thread.

Reply:
reverted